### ISSUE TITLE
Force Download - "The movie with the given tmdbid could not be found!"

### ISSUE THREAD

**Issue by exetico** [Radarr/Radarr#3704]
**Description:**
I'm trying to use the _Force Download_-feature to get a movie.

But every time i'm trying to add an alternative title, it's returning Error 500 to my request.

Message:
```
Request Failed: Error while calling api: Unprocessable Entity
Full error(s): Unprocessable Entity (422, RayId: 81838889-1dd2-4d20-a358-967256e71df2), Details: The movie with the given tmdbid could not be found!
```
Expection:
```
NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIException: Error while calling api: Unprocessable Entity
Full error(s): Unprocessable Entity (422, RayId: 81838889-1dd2-4d20-a358-967256e71df2), Details: The movie with the given tmdbid could not be found!
  at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.Execute[T] (NzbDrone.Common.Http.HttpRequest request) [0x0004f] in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:77 
  at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.AddNewAlternativeTitle (NzbDrone.Core.Movies.AlternativeTitles.AlternativeTitle title, System.Int32 TmdbId) [0x0007e] in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:159 
  at NzbDrone.Api.Movies.AlternativeTitleModule.AddTitle (NzbDrone.Api.Movies.AlternativeTitleResource altTitle) [0x00019] in C:\projects\radarr-usby1\src\NzbDrone.Api\Movies\AlternativeTitleModule.cs:44 
  at NzbDrone.Api.REST.RestModule`1[TResource].<set_CreateResource>b__41_0 (System.Object options) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Api\REST\RestModule.cs:161 
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)
  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 

```

This is kinda the same as #2384 and many other issues, with almost the same issue.

Here's the same response (but in JSON-format) as received in the browser:
```
{
  "message": "Error while calling api: Unprocessable Entity\nFull error(s): Unprocessable Entity (422, RayId: e7185cfb-6529-487c-b2a3-7caa1e8b3ca1), Details: The movie with the given tmdbid could not be found!",
  "description": "NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIException: Error while calling api: Unprocessable Entity\nFull error(s): Unprocessable Entity (422, RayId: e7185cfb-6529-487c-b2a3-7caa1e8b3ca1), Details: The movie with the given tmdbid could not be found!\n  at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.Execute[T] (NzbDrone.Common.Http.HttpRequest request) [0x0004f] in C:\\projects\\radarr-usby1\\src\\NzbDrone.Core\\MetadataSource\\RadarrAPI\\RadarrAPIClient.cs:77 \n  at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.AddNewAlternativeTitle (NzbDrone.Core.Movies.AlternativeTitles.AlternativeTitle title, System.Int32 TmdbId) [0x0007e] in C:\\projects\\radarr-usby1\\src\\NzbDrone.Core\\MetadataSource\\RadarrAPI\\RadarrAPIClient.cs:159 \n  at NzbDrone.Api.Movies.AlternativeTitleModule.AddTitle (NzbDrone.Api.Movies.AlternativeTitleResource altTitle) [0x00019] in C:\\projects\\radarr-usby1\\src\\NzbDrone.Api\\Movies\\AlternativeTitleModule.cs:44 \n  at NzbDrone.Api.REST.RestModule`1[TResource].<set_CreateResource>b__41_0 (System.Object options) [0x00000] in C:\\projects\\radarr-usby1\\src\\NzbDrone.Api\\REST\\RestModule.cs:161 \n  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)\n  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 "
}
```

**Radarr Version:**
 Radarr Ver. 0.2.0.1391 - nightly (Running in a Docker-container with `linuxserver/radarr:nightly`

**Mono Version:**
5.20.1.34 (tarball Tue Jul 16 22:52:32 UTC 2019)

**Debug Logs:**
```
19-8-19 16:30:24.1|Fatal|NzbDroneErrorPipeline|Request Failed

[v0.2.0.1391] NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIException: Error while calling api: Unprocessable Entity
Full error(s): Unprocessable Entity (422, RayId: e7185cfb-6529-487c-b2a3-7caa1e8b3ca1), Details: The movie with the given tmdbid could not be found!
  at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.Execute[T] (NzbDrone.Common.Http.HttpRequest request) [0x0004f] in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:77 
  at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.AddNewAlternativeTitle (NzbDrone.Core.Movies.AlternativeTitles.AlternativeTitle title, System.Int32 TmdbId) [0x0007e] in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:159 
  at NzbDrone.Api.Movies.AlternativeTitleModule.AddTitle (NzbDrone.Api.Movies.AlternativeTitleResource altTitle) [0x00019] in C:\projects\radarr-usby1\src\NzbDrone.Api\Movies\AlternativeTitleModule.cs:44 
  at NzbDrone.Api.REST.RestModule`1[TResource].<set_CreateResource>b__41_0 (System.Object options) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Api\REST\RestModule.cs:161 
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)
  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 

19-8-19 16:30:24.1|Debug|Api|[POST] /api/alttitle: 500.InternalServerError (636 ms)
```

The title is something like:
`Poelsemanden.2018.NORDiC.1080p.WEB-DL.H.264-VendorName`
where the movie is called `Pølsemanden` in Radarr+TMDb.

The Message in the Force Download is:
```
The title "Poelsemanden" could not be found amongst the alternative titles of the movie. This could lead to problems when Radarr wants to import your movie. If you click force download below, the title will be added to the alternative titles using the language selected below.
```

I guess it could be something related to the Danish characters: `æ`, `ø` and `å`. However, it's working from time to time. I just tested with another title where both `æ` and `ø` was in the title. It worked, with no issues, what so ever.

**Comment by nathank1989** [comment]
I have this issue as well. 

```
NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIException: Error while calling api: Unprocessable Entity
Full error(s): Unprocessable Entity (422, RayId: afefdb8a-5082-4550-85ce-9141f360bcaa), Details: The movie with the given tmdbid could not be found!
  at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.Execute[T] (NzbDrone.Common.Http.HttpRequest request) [0x0004f] in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:77 
  at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.AddNewAlternativeTitle (NzbDrone.Core.Movies.AlternativeTitles.AlternativeTitle title, System.Int32 TmdbId) [0x0007e] in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:159 
  at NzbDrone.Api.Movies.AlternativeTitleModule.AddTitle (NzbDrone.Api.Movies.AlternativeTitleResource altTitle) [0x00019] in C:\projects\radarr-usby1\src\NzbDrone.Api\Movies\AlternativeTitleModule.cs:44 
  at NzbDrone.Api.REST.RestModule`1[TResource].<set_CreateResource>b__41_0 (System.Object options) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Api\REST\RestModule.cs:161 
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)
  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 

```

```
Request Failed: Error while calling api: Unprocessable Entity
Full error(s): Unprocessable Entity (422, RayId: afefdb8a-5082-4550-85ce-9141f360bcaa), Details: The movie with the given tmdbid could not be found!
```

How can I fix this?
Version
0.2.0.1358
Mono Version
5.10.1.47 (5.10.1.47 Wed Aug 22 04:26:22 UTC 2018)

**Comment by fryfrog** [comment]
Could you turn logging up to trace and reproduce this error, include a bunch of context from before and after it?

**Comment by LeoShivas** [comment]
Hi,
the bug is still there.

### Here's my version :
![image](https://user-images.githubusercontent.com/28844182/71728381-37aa1e80-2e3d-11ea-8b12-6d6a79e986e5.png)

### Here's how I produce the bug :
![image](https://user-images.githubusercontent.com/28844182/71728542-a2f3f080-2e3d-11ea-8b79-c810abbc727a.png)

### And here's the logs :
**Message**
`Request Failed: Movie with ID 0 does not exist`
**Exception**
```
NzbDrone.Core.Datastore.ModelNotFoundException: Movie with ID 0 does not exist
  at NzbDrone.Core.Datastore.BasicRepository`1[TModel].Get (System.Int32 id) [0x0003d] in C:\projects\radarr-usby1\src\NzbDrone.Core\Datastore\BasicRepository.cs:96 
  at NzbDrone.Core.Movies.MovieService.GetMovie (System.Int32 movieId) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Movies\MovieService.cs:139 
  at NzbDrone.Api.Movies.AlternativeTitleModule.AddTitle (NzbDrone.Api.Movies.AlternativeTitleResource altTitle) [0x00007] in C:\projects\radarr-usby1\src\NzbDrone.Api\Movies\AlternativeTitleModule.cs:43 
  at NzbDrone.Api.REST.RestModule`1[TResource].<set_CreateResource>b__41_0 (System.Object options) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Api\REST\RestModule.cs:161 
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)
  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 

```

**Comment by LeoShivas** [comment]
After clearing the logs, I've increased the log level to trace.

Here's the logs when I make a manual search before the bug :
![image](https://user-images.githubusercontent.com/28844182/71728959-dbe09500-2e3e-11ea-9fe5-cfe19a5e17c8.png)

When I click on the release I want, that produces no log.

Then, here's the log when I click on ''Force download'' :
**Message**
`Request Failed: Movie with ID 0 does not exist`
**Exception**
```
NzbDrone.Core.Datastore.ModelNotFoundException: Movie with ID 0 does not exist
  at NzbDrone.Core.Datastore.BasicRepository`1[TModel].Get (System.Int32 id) [0x0003d] in C:\projects\radarr-usby1\src\NzbDrone.Core\Datastore\BasicRepository.cs:96 
  at NzbDrone.Core.Movies.MovieService.GetMovie (System.Int32 movieId) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Movies\MovieService.cs:139 
  at NzbDrone.Api.Movies.AlternativeTitleModule.AddTitle (NzbDrone.Api.Movies.AlternativeTitleResource altTitle) [0x00007] in C:\projects\radarr-usby1\src\NzbDrone.Api\Movies\AlternativeTitleModule.cs:43 
  at NzbDrone.Api.REST.RestModule`1[TResource].<set_CreateResource>b__41_0 (System.Object options) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Api\REST\RestModule.cs:161 
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)
  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 
```

This is the only entry I have in the log. It seems trace log level doesn't provide more information :-(

**Comment by ta264** [comment]
Trace logs don't show in the UI,you need to look at the actual log files.

**Comment by LeoShivas** [comment]
@ta264, ok. I found them.

I've cleaned them before reproducing the bug.
[logs_radarr.zip](https://github.com/Radarr/Radarr/files/4019773/logs_radarr.zip)

**Comment by kevinvincent** [comment]
Also seeing this issue for the movie Miss Americana

**Comment by jbrody17** [comment]
I am currently running into this issue with Miss Americana.

Steps to reproduce:
- manual search for movie [Miss Americana (2020)]
- click force download button (yellow download icon) on release I want
- select alternative title language from dropdown
- click force download button (blue “Force Download” button)
- loading icon appears and disappears, but force download modal does not disappear

upon checking the UI logs, I see this error:
Request Failed: Error while calling api: Unprocessable Entity Full error(s): Unprocessable Entity (422, RayId: c9e72c08-72f3-4f3e-ab38-8b44856d4c60), Details: The movie with the given tmdbid could not be found!

Full Trace Logs:

[Trace] Http: Res: 304 [GET] /api/release?movieId=1796&sort_by=releaseWeight&order=asc: 200.OK (839 ms)
[Debug] Api: [GET] /api/release?movieId=1796&sort_by=releaseWeight&order=asc: 200.OK (839 ms)
[Trace] Scheduler: Pending Tasks: 0
[Trace] Http: Req: 305 [POST] /api/alttitle
[Trace] Owin: SQLite warning (284): automatic index on MovieFiles(MovieId)


[Trace] HttpClient: Req: [GET] https://api.radarr.video/v2/mappings/add?tmdbid=653567&type=title&language=en&aka_title=Taylor%20Swift%20Miss%20Americana
[Trace] ConfigService: Using default config value for 'proxybypassfilter' defaultValue:''
[Trace] ConfigService: Using default config value for 'proxyusername' defaultValue:''
[Trace] ConfigService: Using default config value for 'proxypassword' defaultValue:''
[Trace] HttpClient: Res: [GET] https://api.radarr.video/v2/mappings/add?tmdbid=653567&type=title&language=en&aka_title=Taylor%20Swift%20Miss%20Americana: 422.422
{"errors":[{"id":"2d9bb13a-f381-4685-a210-44c50633d6eb","status":422,"title":"Unprocessable Entity","detail":"The movie with the given tmdbid could not be found!"}]} (308 ms)
[Trace] NzbDroneErrorPipeline: Handling Exception
[Fatal] NzbDroneErrorPipeline: Request Failed


[v0.2.0.1450] NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIException: Error while calling api: Unprocessable Entity

Full error(s): Unprocessable Entity (422, RayId: 2d9bb13a-f381-4685-a210-44c50633d6eb), Details: The movie with the given tmdbid could not be found!

at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.Execute[T] (NzbDrone.Common.Http.HttpRequest request) [0x0004f] in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:77
at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.AddNewAlternativeTitle (NzbDrone.Core.Movies.AlternativeTitles.AlternativeTitle title, System.Int32 TmdbId) [0x0007e] in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:159
at NzbDrone.Api.Movies.AlternativeTitleModule.AddTitle (NzbDrone.Api.Movies.AlternativeTitleResource altTitle) [0x00019] in C:\projects\radarr-usby1\src\NzbDrone.Api\Movies\AlternativeTitleModule.cs:44
at NzbDrone.Api.REST.RestModule`1[TResource].<set_CreateResource>b__41_0 (System.Object options) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Api\REST\RestModule.cs:161
at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)
at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0

[Trace] Http: Res: 305 [POST] /api/alttitle: 500.InternalServerError (417 ms)
[Debug] Api: [POST] /api/alttitle: 500.InternalServerError (417 ms)
[Trace] Http: Req: 306 [GET] /api/config/naming
[Trace] Http: Res: 306 [GET] /api/config/naming: 200.OK (8 ms)
[Debug] Api: [GET] /api/config/naming: 200.OK (8 ms)
[Trace] Http: Req: 307 [GET] /api/config/indexer
[Trace] Http: Req: 308 [GET] /api/config/mediamanagement

**Comment by jbrody17** [comment]
@fryfrog any thoughts on this?

**Comment by arebokert** [comment]
Issue does still remains, movie in question is Star Wars 4K77. Exactly the same problem as previously posted logs.

**Comment by splitice** [comment]
Still exists with "Interpreters (2019)". This is going to be a very common issue with the Coronavirus delaying most movies.

**Comment by sohlican** [comment]
This issue is not resolved.  I have encountered it today.  Pets United (2019)

Details
Message
Request Failed: Error while calling api: Unprocessable Entity
Full error(s): Unprocessable Entity (422, RayId: 92f9e0ae-be64-4f1e-8324-64d6962eaf05), Details: The movie with the given tmdbid could not be found!

Exception
NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIException: Error while calling api: Unprocessable Entity
Full error(s): Unprocessable Entity (422, RayId: 92f9e0ae-be64-4f1e-8324-64d6962eaf05), Details: The movie with the given tmdbid could not be found!
   at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.Execute[T](HttpRequest request) in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:line 70
   at NzbDrone.Core.MetadataSource.RadarrAPI.RadarrAPIClient.AddNewAlternativeYear(Int32 year, Int32 tmdbId) in C:\projects\radarr-usby1\src\NzbDrone.Core\MetadataSource\RadarrAPI\RadarrAPIClient.cs:line 167
   at NzbDrone.Api.Movies.AlternativeYearModule.AddYear(AlternativeYearResource altYear) in C:\projects\radarr-usby1\src\NzbDrone.Api\Movies\AlternativeYearModule.cs:line 46
   at NzbDrone.Api.REST.RestModule`1.<set_CreateResource>b__41_0(Object options) in C:\projects\radarr-usby1\src\NzbDrone.Api\REST\RestModule.cs:line 161
   at CallSite.Target(Closure , CallSite , Func`2 , Object )
   at Nancy.Routing.Route.<>c__DisplayClass4.<Wrap>b__3(Object parameters, CancellationToken context)
Radarr Ver. 0.2.0.1504
develop

**Comment by ta264** [comment]
Not going to be fixed on 0.2. You can try the V3 alpha if it's a big issue

**Comment by gaLDWoRE** [comment]
Still on 0.2.0.1504, Same issue with Pets United

**Comment by bakerboy448** [comment]
Never ever will be fixed on 0.2. You can try the V3 beta if it's a big issue

**Comment by gaLDWoRE** [comment]
> Never ever will be fixed on 0.2. You can try the V3 beta if it's a big issue

When will V3 come out of beta?
I'm running it on Unraid, not sure how to install Betas and if it's a good idea.

**Comment by bakerboy448** [comment]
> When will V3 come out of beta?

When it's done. sonarr v3 has been in beta for a few years

> I'm running it on Unraid, not sure how to install Betas and if it's a good idea.

Assuming docker, change your docker tag. But backup your db first
